RoddyCore/src/de/dkfz/roddy/config/ToolEntry.java

```diff
@@ -227,15 +262,15 @@ public ToolStringParameter clone() {
         public final List<ToolConstraint> constraints;
         public final String scriptParameterName;
         public final String filenamePatternSelectionTag;
-        public final boolean checkFile;
+        public final ToolFileParameterCondition checkFile;
         public final String parentVariable;
         private List<ToolFileParameter> childFiles;
 
-        public ToolFileParameter(Class<BaseFile> fileClass, List<ToolConstraint> constraints, String scriptParameterName, boolean checkFile) {
+        public ToolFileParameter(Class<BaseFile> fileClass, List<ToolConstraint> constraints, String scriptParameterName, ToolFileParameterCondition checkFile) {
             this(fileClass, constraints, scriptParameterName, checkFile, FilenamePattern.DEFAULT_SELECTION_TAG, null, null);
         }
 
-        public ToolFileParameter(Class<BaseFile> fileClass, List<ToolConstraint> constraints, String scriptParameterName, boolean checkFile, String filenamePatternSelectionTag, List<ToolFileParameter> childFiles, String parentVariable) {
+        public ToolFileParameter(Class<BaseFile> fileClass, List<ToolConstraint> constraints, String scriptParameterName, ToolFileParameterCondition checkFile, String filenamePatternSelectionTag, List<ToolFileParameter> childFiles, String parentVariable) {
```

Comment:
It does not necessarily need to be pure boolean. It can also hold a condition and will default to check when something fails.